### PR TITLE
MultipartPost test now uses the form feature

### PR DESCRIPTION
**Pull request by DavidObando**
The previous implementation relied on manually reading the body and using a MultipartReader object in order to fully consume all the parts and read all the bytes. This is actually redundant as ASP.NET has a built-in feature where multiparts are read and cached in a local file even before the controller is called. All the controller has to do is either specify a complex object as its parameter or access the files via the Request.Form object.

### CODE REVIEW

**Comment by SajayAntony** [comment]
:shipit:

File: testapp/MultipartPOST/MultipartPOST/Controllers/UploadController.cs [hunk 1]
```diff
@@ -15,7 +15,7 @@ public class UploadController : Controller
     {
         private const int DefaultBufferSize = 4096;
```
Comment (davidfowl) [inline testapp/MultipartPOST/MultipartPOST/Controllers/UploadController.cs]: Can delete?

File: testapp/MultipartPOST/MultipartPOST/Controllers/UploadController.cs [hunk 2]
```diff
@@ -24,27 +24,11 @@ public class UploadController : Controller
 
             PrintLine("Processing a POST request to /api/upload");
 
-            var buffer = new byte[DefaultBufferSize];
-            var boundary = GetBoundary(Request.ContentType);
-            var reader = new MultipartReader(boundary, Request.Body);
-            while (true)
+            var form = Request.Form;
```
Comment (davidfowl) [inline testapp/MultipartPOST/MultipartPOST/Controllers/UploadController.cs]: Is this synchronous on purpose?

File: testapp/MultipartPOST/MultipartPOST/Controllers/UploadController.cs (hunk 1 above)
Comment (DavidObando) [inline testapp/MultipartPOST/MultipartPOST/Controllers/UploadController.cs]: Will do.

File: testapp/MultipartPOST/MultipartPOST/Controllers/UploadController.cs (hunk 2 above)
Comment (DavidObando) [inline testapp/MultipartPOST/MultipartPOST/Controllers/UploadController.cs]: I'm not using any async APIs in the body of this method.

Comment (davidfowl) [inline testapp/MultipartPOST/MultipartPOST/Controllers/UploadController.cs]: I'm mostly asking if this test is purposely testing the sync API vs the async form reading API.

Comment (DavidObando) [inline testapp/MultipartPOST/MultipartPOST/Controllers/UploadController.cs]: I see. I hadn't realized there's an async API for form reading. I'll take a look and report back. Thanks for the observation.

**Pull request merged**